lms/views/lti/deeplinking.py

```diff
@@ -58,11 +68,83 @@ def content_item_selection(context, request):
-            "lti_version": request.params["lti_version"],
+            "lti_version": request.parsed_params["lti_version"],
         },
     )
 
     return {}
+
+
+class DeeplinkingFieldsRequestSchema(JSONPyramidRequestSchema):
+    deep_linking_settings = fields.Dict(required=False)
+    content_item_return_url = fields.Str(required=True)
+    content = fields.Dict(required=True)
+
+    extra_params = fields.Dict(required=False)
+
+
+@view_defaults(
+    request_method="POST", renderer="json", schema=DeeplinkingFieldsRequestSchema
+)
+class DeeplinkingFieldsViews:
+    """
+    Views that return the required form fields to complete a deeplinking request to the LMS.
+
+    After the user picks a document to be annotated the frontend will call these views
+    to get all the necessary fields to configure the assignment submitting them in a form to the LMS.
+    """
+
+    def __init__(self, request):
+        self.request = request
+
+        self.application_instance = self.request.find_service(
+            name="application_instance"
+        ).get_current()
+
+    @view_config(route_name="lti.deeplinking.formfields")
+    def formfields_v13(self):
+        url = self._content_to_url(
+            self.request.parsed_params["content"],
+            self.request.parsed_params.get("extra_params"),
+        )
+
+        ltia_signature = self.request.find_service(LTIAHTTPService).sign(
+            {
+                "https://purl.imsglobal.org/spec/lti/claim/deployment_id": self.application_instance.deployment_id,
+                "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiDeepLinkingResponse",
+                "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+                "https://purl.imsglobal.org/spec/lti-dl/claim/content_items": [
+                    {"type": "ltiResourceLink", "url": url}
+                ],
+                "https://purl.imsglobal.org/spec/lti-dl/claim/data": self.request.parsed_params[
+                    "deep_linking_settings"
+                ],
+            }
+        )
+        # In LTI1.3 there's just one `JWT` field which includes all the necessary information
+        return {"JWT": ltia_signature}
+
+    def _content_to_url(self, content, extra_params):
+        """
+        Translate content information from the frontend to a launch URL.
+
+        We submit the content information to the LMS as an URL pointing to our
+        `lti_launches` endpoint with any information required to identity
+        the content as query parameters.
+        """
+        return_url = self.request.route_url("lti_launches")
+
+        params = dict(extra_params or {})
+
+        content = self.request.parsed_params["content"]
+        if content["type"] == "file":
+            params["canvas_file"] = "true"
+            params["file_id"] = content["file"]["id"]
+        else:
+            assert content["type"] == "url", "Unknown content type"
+            params["url"] = content["url"]
```

Comment:
That would be ideal and made a note in my post-LTI cleanup for it but I rather to leave it out of this PR as it might involve some changes around all different file pickers in the frontend side.